suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -0,0 +1,165 @@
+
+import { DeviceItem } from './DeviceItem';
+import { useDeviceManager } from '../hooks/useDeviceManager';
+import { MANAGER_MODAL_BOTTOM_RADIUS } from './DeviceManagerModal';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    RootStackParamList,
+    RootStackRoutes.AppTabs,
+    RootStackParamList
+>;
+
+type DeviceListProps = {
+    isVisible: boolean;
+    onSelectedDevice?: (device: AcquiredDevice) => void;
+};
+
+const ITEM_HEIGHT = 62;
+const SPACING = 8;
+const BUTTON_HEIGHT = 48;
+const SEPARATOR_HEIGHT = 18;
+const VERTICAL_PADDING = 16;
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.medium,
+}));
+
+export const DeviceList = ({ isVisible, onSelectedDevice }: DeviceListProps) => {
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+    const { setIsDeviceManagerVisible } = useDeviceManager();
+    const device = useSelector(selectDevice);
+    const physicalDevices = useSelector(selectPhysicalDevices);
+    const portfolioDevice = useSelector((state: DeviceRootState) =>
+        selectDeviceById(state, PORTFOLIO_TRACKER_DEVICE_ID),
+    );
+    const { utils } = useNativeStyles();
+    const opacity = useSharedValue(0);
+    const height = useSharedValue(0);
+    const [isShown, setIsShown] = useState(false);
+
+    const allDevices = [...physicalDevices, portfolioDevice].filter(
+        a => a !== undefined,
+    ) as TrezorDevice[];
+    const otherDevices = getSortedDevicesWithoutInstances(allDevices, device?.id);
+
+    const handleConnectDevice = () => {
+        if (device) {
+            onSelectedDevice?.(device as AcquiredDevice);
+        }
+        setIsDeviceManagerVisible(false);
+        navigation.navigate(RootStackRoutes.ConnectDeviceStack, {
+            screen: ConnectDeviceStackRoutes.ConnectAndUnlockDevice,
+        });
+        analytics.report({
+            type: EventType.DeviceManagerClick,
+            payload: { action: 'connectDeviceButton' },
+        });
+    };
+
+    //to hide/show the list with animation
+    useEffect(() => {
+        let timer: ReturnType<typeof setTimeout>;
+
+        if (isVisible) {
+            const h =
+                otherDevices.length * (ITEM_HEIGHT + SPACING) +
+                BUTTON_HEIGHT +
+                MANAGER_MODAL_BOTTOM_RADIUS * 2 +
+                (otherDevices.length > 0 ? SEPARATOR_HEIGHT : 0) +
+                2 * VERTICAL_PADDING;
+            height.value = withTiming(h, { duration: 300 });
+            opacity.value = withDelay(100, withTiming(1, { duration: 200 }));
+            setIsShown(isVisible);
+        } else {
+            height.value = withDelay(100, withTiming(0, { duration: 200 }));
+            opacity.value = withTiming(0, { duration: 300 });
+            timer = setTimeout(() => {
+                setIsShown(isVisible);
+            }, 300);
+        }
+
+        return () => clearTimeout(timer);
```

Comment:
No need for an extra timer. You can attach a callback to the previous `withTiming` call which will be executed right after the animation finishes. https://docs.swmansion.com/react-native-reanimated/docs/animations/withTiming/#callback-

Reply:
5825a46c34